Makefile

```diff
@@ -1,6 +1,8 @@
 PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 CWD := $(shell pwd)
 
+GO ?= go
```

Comment:
using different versions can cause unwanted confusion (for example go.mod/go.sum gets generated differently with 1.16 and 1.17) . best option is to stick to the required versions

Reply:
Yes, but the default behavior will remain the same nonetheless. But sometimes we might need to test out multiple golang executable, usually when updating dependencies.

Reply:
Can't we temporarily set PATH and test it?